[Chua Sue-Ann] iP

sueann-chua opened the pull request:
# Duke 

Duke helps manage your tasks including todo-type tasks, deadlines and events that you need to keep track of. It's: 

* text-based
* easy to use 
* ~~FAST~~ SUPER FAST to use
* **FREE**

All you need to do is,
1. Download it [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/279)
1. Run it
1. Add your tasks using the command words todo, deadline, event
1. Let it manage tasks for you

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [x] Compile list of tasks
- [x] Mark tasks as done
- [x] Delete tasks from list
- [x] Find tasks using keywords

Feel free to take a look at the code! :star_struck: Here's a snippet of the `main` method:
```
public static void main(String[] args) throws Exception {
        Files.createDirectories(Paths.get("data/"));
        File dukeFile = new File("data/duke.txt");
        new Duke(dukeFile).run();
    }
```

Commit:
Add Gradle support

sueann-chua committed:
level 1: greet, echo, exit

sueann-chua committed:
Level-2: Add, list

sueann-chua committed:
Level-3: Mark as Done

sueann-chua committed:
Level-4: ToDos, Events, Deadlines

sueann-chua committed:
A-TextUiTesting

sueann-chua committed:
Level-5: Handle errors

sueann-chua committed:
Level-6: Delete

sueann-chua committed:
Final edits: switch

sueann-chua committed:
Add toString methods

sueann-chua committed:
Add level 7 save feature

sueann-chua committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Task.java

sueann-chua committed:
Add date and time for deadlines for Level 8

sueann-chua committed:
Merge branch 'branch-Level-8'

sueann-chua committed:
Add Storage class

sueann-chua committed:
Add Parser class

sueann-chua committed:
Add TaskList class

sueann-chua committed:
Add Ui class

sueann-chua committed:
Divide classes into packages

sueann-chua committed:
Add JUnit tests

sueann-chua committed:
Add javadocs to half

sueann-chua committed:
Edit code to comply with coding standard

sueann-chua committed:
Add JavaDoc to Storage and Ui files

sueann-chua committed:
Add javadocs for remaining classes

sueann-chua committed:
Edit code to comply with coding standards

sueann-chua committed:
Add find function

sueann-chua committed:
Merge branch 'branch-Level-9'

KishendranVendarKon commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,152 @@
+        String command = Parser.getCommand(input);
+        String description = Parser.getDescription(input);
+
+        switch (command) {
```
The case blocks should not be indented based on this modules coding standard.

KishendranVendarKon commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Getter to get the arraylist of tasks
```
Method headers should have a "." at the end of the sentence. I noticed the same issue a couple of times.

KishendranVendarKon commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,200 @@
+    /**
+     * Method to show message to warn user that task number is missing
+     */
+    public static void warningMissingNumber() {
```
Function should be named as a verb.

KishendranVendarKon commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,200 @@
+        showToUser(LINE, successfulMessage);
+    }
+
+    /**
```
There should be a space between the description and the @param in header comments. I noticed this issues a couple other times.

KishendranVendarKon commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,35 @@
+import duke.tasks.Todo;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Static imports should be before all other import statements.

KishendranVendarKon reviewed (commented):
Looks good. Just some cosmetic changes requested.

sueann-chua committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

sueann-chua committed:
Add Launcher class and edited build.gradle file

sueann-chua committed:
Create GUI for Duke

sueann-chua committed:
Echo user and add dialog boxes

sueann-chua committed:
Edit GUI to interact with user

sueann-chua committed:
Use FXML

sueann-chua committed:
Add parser method to parse user input and generate response

sueann-chua committed:
Add parser to take in user input and generate response for all commands

sueann-chua committed:
Add manifest file

sueann-chua committed:
Amended javadocs to comply with coding standards

Gordon25 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,82 @@
+ */
+public class Duke {
+
+    private duke.Storage storage;
+    private duke.TaskList taskList;
```
Perhaps you want to consider importing duke.Storage and duke.TaskList classes above so you can simply refer to the class names? 🤔

Gordon25 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,82 @@
+        try {
+            File dukeFile = new File("data/duke.txt");
+            PrintWriter writer  = this.storage.load();
+
+            Ui.showWelcomeMessage();
+
```
I like that you use space line, improves the readability of your code. 👍

Gordon25 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,201 @@
+                String oldDescription = taskList.getIndividualTask(taskNum - 1).toString();
+                taskList.completeTask(taskNum); //todo
+                Storage.saveAsCompleted(file, taskList.getIndividualTask(taskNum - 1), oldDescription); //todo
+            } else if (Parser.getCommand(action).equals("todo")
+                    || Parser.getCommand(action).equals("deadline")
+                    || Parser.getCommand(action).equals("event")) { // add task to to-do list
```
I like your use of breaks 😃

Gordon25 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * String representation of a task instance.
```
Could be phrased similar to like gets ... ... ?

Gordon25 reviewed (commented):
Overall, LGTM!

sueann-chua committed:
Fix bug where data was not being written into the duke.txt file.

sueann-chua committed:
Javadocs: complete all javadocs fields such as @param.

Javadocs should comply with coding standards.

All @param fields and @return fields have been completed and filled in.

sueann-chua committed:
Remove getTaskType methods in Deadline and Event.

The getTaskType method was not being used anywhere in the code. And the getTaskType method was previously in Task class and was overridden in child classes such as Deadline and Event.

Removing the getTaskType method from child classes of Task class helps clean up the code.

sueann-chua committed:
Add Java assertions to Deadline and Event classes.

Deadline and Event classes have a few methods that assume the user inputs follow the correct format such as 'deadline {description} /by yyyy-mm-dd time' and 'event {description} /at {details}'.

Adding Java assertions help define these assumptions and help in testing.

sueann-chua committed:
Edited code to improve code quality.

There was dead code that was removed, and if else block in parser was changed to switch block for readability. Magic numbers in deadline and event classes were also removed.

sueann-chua committed:
Merge pull request #1 from sueann-chua/branch-A-Assertions

Add Java assertions to Deadline and Event classes.

sueann-chua committed:
Merge branch 'master' of https://github.com/sueann-chua/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/tasks/Deadline.java
#	src/main/java/duke/tasks/Event.java

sueann-chua committed:
Merge pull request #2 from sueann-chua/branch-A-CodeQuality

Edit code to improve code quality.

sueann-chua committed:
Add help page feature

sueann-chua committed:
Remove unnecessary todo comment

sueann-chua committed:
Improve GUI. Changed background colour to white and added padding for cleaner look.

sueann-chua committed:
Improve GUI such that if text is too long, it will not get cut off by ellipsis.

sueann-chua committed:
Remove unnecessary import statements

sueann-chua committed:
Add representative screenshot of product

sueann-chua committed:
Add additional instruction to exit app

sueann-chua committed:
Add user guide to README

sueann-chua committed:
Set theme jekyll-theme-tactile

sueann-chua committed:
Set theme jekyll-theme-cayman

sueann-chua committed:
Remove ### from notes as it shows ### on the github page

sueann-chua committed:
Merge branch 'master' of https://github.com/sueann-chua/ip

sueann-chua committed:
Tweak code according to suggestions from reviewers

sueann-chua committed:
Add method header for gethelpMessage method

sueann-chua committed:
Tweaked code to exit program when user types 'bye'.

sueann-chua committed:
Edit method header descriptions to comply with coding standards

sueann-chua committed:
Remove or repeated code

sueann-chua committed:
Improve code quality

sueann-chua committed:
Add method header for DialogBox methods

sueann-chua committed:
Delete dead code
